src/mame/heathkit/tlb.cpp

```diff
@@ -242,16 +277,18 @@ uint8_t heath_tlb_device::kbd_flags_r()
 
 int heath_tlb_device::mm5740_shift_r()
 {
-	return ((m_kbspecial->read() ^ 0x120) & 0x120) ? ASSERT_LINE : CLEAR_LINE;
+	return ((m_kbspecial->read() & 0x120) != 0x120) ? ASSERT_LINE : CLEAR_LINE;
 }
 
 int heath_tlb_device::mm5740_control_r()
 {
-	return ((m_kbspecial->read() ^ 0x10) & 0x10) ? ASSERT_LINE: CLEAR_LINE;
+	return (m_kbspecial->read() & 0x10) ? CLEAR_LINE : ASSERT_LINE;
 }
 
 void heath_tlb_device::mm5740_data_ready_w(int state)
 {
+	check_for_reset();
+
```

Comment:
Is this when the reset key combination is actually checked, or does it happen out-of-band?  If it happens out-of-band, you should attach callbacks to the relevant keys so you get a notification when they’re changed.